homeassistant/components/laundrify/__init__.py

```diff
@@ -0,0 +1,61 @@
+"""The laundrify integration."""
+from __future__ import annotations
+
+from laundrify_aio import LaundrifyAPI
+from laundrify_aio.exceptions import ApiConnectionException, UnauthorizedException
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_ACCESS_TOKEN, Platform
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryAuthFailed, ConfigEntryNotReady
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+
+from .const import DEFAULT_POLL_INTERVAL, DOMAIN
+from .coordinator import LaundrifyUpdateCoordinator
+
+PLATFORMS = [Platform.BINARY_SENSOR]
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up laundrify from a config entry."""
+
+    session = async_get_clientsession(hass)
+    api_client = LaundrifyAPI(entry.data[CONF_ACCESS_TOKEN], session)
+
+    try:
+        await api_client.validate_token()
+    except UnauthorizedException as err:
+        raise ConfigEntryAuthFailed("Invalid authentication") from err
+    except ApiConnectionException as err:
+        raise ConfigEntryNotReady("Cannot reach laundrify API") from err
+
+    coordinator = LaundrifyUpdateCoordinator(hass, api_client, DEFAULT_POLL_INTERVAL)
+
+    await coordinator.async_config_entry_first_refresh()
+
+    hass.data.setdefault(DOMAIN, {})[entry.entry_id] = {
+        "api": api_client,
+        "coordinator": coordinator,
+    }
+
+    entry.async_on_unload(entry.add_update_listener(options_update_listener))
+
+    hass.config_entries.async_setup_platforms(entry, PLATFORMS)
+
+    return True
+
+
+async def options_update_listener(
+    hass: HomeAssistant, config_entry: ConfigEntry
+) -> None:
+    """Handle options update."""
+    await hass.config_entries.async_reload(config_entry.entry_id)
```

Comment:
not needed anymore now that the options flow is removed

Reply:
Removed in 95018cedae969e535cba4f82c75d63664d162291